numpy/linalg/linalg.py

```diff
@@ -1933,36 +1934,40 @@ def lstsq(a, b, rcond=-1):
     a, bstar = _fastCopyAndTranspose(t, a, bstar)
     a, bstar = _to_native_byte_order(a, bstar)
     s = zeros((min(m, n),), real_t)
-    nlvl = max( 0, int( math.log( float(min(m, n))/2. ) ) + 1 )
```

Comment:
This was meant to be `log2`, apparently, and that `2` was supposed to be `1 + magic_internal_lapack_number`. Better to let LAPACK calculate it, I think.